src/main/java/core/mvc/asis/DispatcherServlet.java

```diff
@@ -1,53 +1,58 @@
 import javax.servlet.ServletException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.io.IOException;
+import java.util.Optional;
 
 @WebServlet(name = "dispatcher", urlPatterns = "/", loadOnStartup = 1)
 public class DispatcherServlet extends HttpServlet {
     private static final long serialVersionUID = 1L;
     private static final Logger logger = LoggerFactory.getLogger(DispatcherServlet.class);
-    private static final String DEFAULT_REDIRECT_PREFIX = "redirect:";
 
-    private RequestMapping rm;
+    private HandlerMappings handlerMappings;
+    private HandlerAdapters handlerAdapters;
 
     @Override
     public void init() throws ServletException {
-        rm = new RequestMapping();
-        rm.initMapping();
+        handlerMappings = new HandlerMappings();
+        handlerAdapters = new HandlerAdapters();
     }
 
     @Override
-    protected void service(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
-        String requestUri = req.getRequestURI();
-        logger.debug("Method : {}, Request URI : {}", req.getMethod(), requestUri);
+    protected void service(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        String requestUri = request.getRequestURI();
+        logger.debug("Method : {}, Request URI : {}", request.getMethod(), requestUri);
 
-        Controller controller = rm.findController(requestUri);
         try {
-            String viewName = controller.execute(req, resp);
-            move(viewName, req, resp);
+            handleRequest(request, response);
         } catch (Throwable e) {
-            logger.error("Exception : {}", e);
+            logger.error("Exception : {}", e.getMessage(), e);
             throw new ServletException(e.getMessage());
         }
     }
 
-    private void move(String viewName, HttpServletRequest req, HttpServletResponse resp)
-            throws ServletException, IOException {
-        if (viewName.startsWith(DEFAULT_REDIRECT_PREFIX)) {
-            resp.sendRedirect(viewName.substring(DEFAULT_REDIRECT_PREFIX.length()));
+    private void handleRequest(HttpServletRequest request, HttpServletResponse response) throws Exception {
+        Optional<Object> maybeHandler = handlerMappings.getHandler(request);
+        if (maybeHandler.isEmpty()) {
+            response.setStatus(HttpServletResponse.SC_NOT_FOUND);
             return;
         }
 
-        RequestDispatcher rd = req.getRequestDispatcher(viewName);
-        rd.forward(req, resp);
+        Object handler = maybeHandler.get();
+        HandlerAdapter handlerAdapter = handlerAdapters.getHandlerAdapter(handler);
+
+        ModelAndView mav = handlerAdapter.handle(request, response, handler);
+        mav.getView().render(mav.getModel(), request, response);
```

Comment:
```suggestion
        mav.render(mav.getModel(), request, response);
```

view를 꺼내서 렌더링하기 보다 ModelAndView에 메서드를 추가해보면 어떨까요